components/org.wso2.carbon.identity.discovery/src/test/java/org/wso2/carbon/identity/discovery/DefaultOIDCProcessorTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.discovery;
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.discovery.builders.DefaultOIDCProviderRequestBuilder;
+import org.wso2.carbon.identity.discovery.builders.ProviderConfigBuilder;
+import org.wso2.carbon.identity.testutil.IdentityBaseTest;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@PrepareForTest({DefaultOIDCProcessor.class})
+public class DefaultOIDCProcessorTest extends IdentityBaseTest {
+
+    private static final String SUPER_TENANT = "carbon.super";
```

Comment:
It's better to use `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME` here as you have done in the `DefaultOIDCProviderRequestBuilderTest`